### ISSUE TITLE
Use Closure syntax for helix/impl/class.js

### ISSUE THREAD

**Issue by Lokeh** [Lokeh/helix#52]
Currently, we maintain a separate branch for using helix with vanilla CLJS (e.g. figwheel, `:bundle`, etc.): `fix-figwheel`

This is due to `src/helix/impl/class.js` originally being written using ES Module syntax (as that worked well with shadow-cljs), and when I tried to rewrite it using Google Closure syntax it didn't immediately work with shadow-cljs.

Ideally we would align the two branches so that it's less confusing for users.

**Comment by SevereOverfl0w** [comment]
I was able to get this to work with both Shadow & Figwheel by:
- Using the latest figwheel snapshot (0.2.5-snapshot)
- Using Cljs 1.10.764 - this was after they updated google closure
- Using goog.provide.

I tested by:
- Running `clojure -A:user/shadow-cljs:dev watch :dev` for shadow & using the devcards
- Taking a figwheel-main template, bumping figwheel-main, cljs, cljsjs/react, & adding my helix fork.  I then copied the code from the README & confirmed it worked.
